jobs/status.go

```diff
@@ -82,26 +84,40 @@ func StartStatusReporter(ctx context.Context, scanCtx *scan.ScanContext) chan er
 			case <-scanCtx.ScanStatusChan:
 				ts = time.Now()
 			case <-ticker.C:
+				if scanCtx.Stopped.Load() == true {
+					core.GetSession().Log.Error("Scanner job stopped, scan_id: %s", scan_id)
+					break loop
+				}
+
+				counter++
+				//We perform the check once per 30 seconds
+				if counter%30 != 0 {
+					continue
+				}
+
 				elapsed := int(time.Since(ts).Seconds())
 				if elapsed > threshold {
 					err = fmt.Errorf("Scan job aborted due to inactivity")
-					core.GetSession().Log.Error("Scanner job aborted as no update within threshold, Scan id:" + scan_id)
+					core.GetSession().Log.Error("Scanner job aborted due to inactivity, scan_id: %s", scan_id)
 					scanCtx.Aborted.Store(true)
 					break loop
 				} else {
 					writeSecretScanStatus("IN_PROGRESS", scan_id, "")
 				}
 			}
 		}
+
 		if abort != nil {
 			writeSecretScanStatus("CANCELLED", scan_id, abort.Error())
-			return
-		}
-		if err != nil {
+		} else if scanCtx.Stopped.Load() == true {
```

Comment:
I see we have `abort` but it does not seem to be used. Can we remove the `if abort != nil` entirely?
If not, might be good to merge it with `scanCtx.Stopped`

Reply:
We have these 2 flags on the scan job as a signal for 2 different activities:
* **Abort**: This signals the case when the scan has been aborted due to lack of activity. 
We set this in the StatusChecker routine to signal that we need to exit the scan. 
We use this flag in the below to exit the running scan: 
https://github.com/deepfence/SecretScanner/blob/60dfd9c89ab2b01e19fec1ad5650e5a09ad2078e/scan/process_image.go#L825

* **Stopped**: This signals that the scan was stopped by api. 
We set this in the "StopScan" grpc request
We use this flag in the below to exit the running scan.
https://github.com/deepfence/SecretScanner/blob/60dfd9c89ab2b01e19fec1ad5650e5a09ad2078e/scan/process_image.go#L829